### PR TITLE
Introduce built-in mutability entitlements

### PR DESCRIPTION
**Pull request by SupunS**
Work towards: https://github.com/onflow/flips/pull/86

⚠️  Depends on https://github.com/onflow/cadence/pull/2573

## Description

Introduces three built-in mutability-related entitlements:
 -  `Insertable`
 - `Removable`
 - `Mutable`

Note: This PR only introduces these entitlements. Array/dictionary functions still don't use them. That will be added in a follow-up PR.

______

- [ ] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [ ] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

**Review by dsainati1 — APPROVED** [review]
Nice!

File: runtime/tests/checker/entitlements_test.go [hunk 1]
```diff
@@ -5110,3 +5110,25 @@ func TestCheckAttachProvidedEntitlements(t *testing.T) {
 		require.Equal(t, errs[1].(*sema.RequiredEntitlementNotProvidedError).RequiredEntitlement.Identifier, "E")
 	})
 }
+
+func TestCheckBuiltinEntitlements(t *testing.T) {
+
+	t.Parallel()
+
+	_, err := ParseAndCheck(t, `
+        struct S {
```
[the remark this answers is not in the document]
Comment (SupunS) [reply runtime/tests/checker/entitlements_test.go]: Added: a073e17db116a66ed133407acbac37f721b9d0ee

**Review by turbolent — APPROVED** [review]
Nice! Great idea to extend the Go code generator to support entitlement declarations, that will come in handy for the `Account` type refactor 👌

**Comment by turbolent** [comment]
We can probably merge this once the FLIP is approved, even if the names are still going to potentially get adjusted later

**Pull request merged**